pkg/kubecost/allocation_test.go

```diff
@@ -2067,9 +2067,332 @@ func TestAllocationSetRange_Accumulate(t *testing.T) {
 		t.Fatalf("accumulating AllocationSetRange: expected %f minutes; actual %f", 2880.0, alloc.Minutes())
 	}
 }
+func TestAllocationSetRange_AccumulateBy_Nils(t *testing.T) {
```

Comment:
Thank you for writing unit tests! I hope that was useful and instructive.

Gotta run for now, so I'll have a look at all of these tomorrow -- but not terribly worried about having every last case covered. This is great.